tests/unit/s2n_openssl_test.c

```diff
@@ -21,21 +21,33 @@ int main(int argc, char** argv)
 {
     BEGIN_TEST();
 
-    const char* env_libcrypto = getenv("S2N_LIBCRYPTO");
-    if (env_libcrypto == NULL) {
-        END_TEST();
-    }
+    /* Confirm "S2N_LIBCRYPTO" env variable matches the linked libcrypto. */
+    {
+        const char* env_libcrypto = getenv("S2N_LIBCRYPTO");
+        if (env_libcrypto == NULL) {
+            END_TEST();
+        }
 
-    if (strcmp(env_libcrypto, "boringssl") == 0) {
-        EXPECT_FALSE(s2n_libcrypto_is_awslc());
-        EXPECT_TRUE(s2n_libcrypto_is_boringssl());
-    } else if (strstr(env_libcrypto, "awslc") != NULL) {
-        EXPECT_TRUE(s2n_libcrypto_is_awslc());
-        EXPECT_FALSE(s2n_libcrypto_is_boringssl());
-    } else {
-        EXPECT_FALSE(s2n_libcrypto_is_awslc());
-        EXPECT_FALSE(s2n_libcrypto_is_boringssl());
-    }
+        if (strcmp(env_libcrypto, "boringssl") == 0) {
+            EXPECT_FALSE(s2n_libcrypto_is_awslc());
+            EXPECT_TRUE(s2n_libcrypto_is_boringssl());
+        } else if (strstr(env_libcrypto, "awslc") != NULL) {
+            EXPECT_TRUE(s2n_libcrypto_is_awslc());
+            EXPECT_FALSE(s2n_libcrypto_is_boringssl());
+        } else {
+            EXPECT_FALSE(s2n_libcrypto_is_awslc());
+            EXPECT_FALSE(s2n_libcrypto_is_boringssl());
+        }
```

Comment:
Could we make this an exclusive match?

I think it'd be nice to have a case where we have some sort of assertion on `s2n_libcrypto_is_openssl` returning true.